### ISSUE TITLE
Link.equals is broken (A->B == A->C)

### ISSUE THREAD

**Issue by drekbour** [nidi3/graphviz-java#96]
[Link.java](../blob/master/graphviz-java/src/main/java/guru/nidi/graphviz/model/Link.java) de-duplicates itself incorrectly.

`MutableGraph.add(MutableNode)` with (A -> B) and (A -> C) de-duplicates the second call because `Link` ignores its target and these two are deemed equal.
`MutableGraph.add(MutableNode)` with (A -> {B,C}) and (A -> D) works because the count is differs !

**Comment by drekbour** [comment]
FYI, workaround is to add a unique attribute to each link
````java
mutableNode.addLink(to(node(tgt))
                .with("constraint", "false")
                .with("https://github.com/nidi3/graphviz-java/issues/96", tgt)
````

**Comment by nidi3** [comment]
This was because doing `equals` in a circular data structure is tricky ;)
Finally fixed in `0.12.1`
